public/controller/data_functions.php

```diff
@@ -353,6 +391,25 @@ function is_valid_zip_size($filename, $max_size) {
     return true;
 }
 
+function get_zip_size($filename) {
+    $size = 0;
+    $zip = zip_open($filename);
+    //
+    // FIXME: bug with this check?  Getting this warning when a non zip is uploaded
+    //
+    // Warning: zip_read() expects parameter 1 to be resource, integer given
+    // Warning: zip_close() expects parameter 1 to be resource, integer given
+    //
+
+    if (is_resource($zip)) {
+        while ($inner_file = zip_read($zip)) {
+            $size += zip_entry_filesize($inner_file);
+        }
+    }
+      zip_close($zip);
```

Comment:
you should only be using ```zip_close``` if you have a zip resource, hence it should be in the if statement (as else you run the risk of getting a warning if zip_open returns an integer error code).